flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -171,8 +171,9 @@ public void open(Configuration parameters) throws Exception {
 		super.open(parameters);
 
 		// check and pass the configuration properties
-		KinesisProducerConfiguration producerConfig = KinesisConfigUtil.validateProducerConfiguration(configProps);
+		KinesisProducerConfiguration producerConfig = KinesisConfigUtil.getValidatedProducerConfiguration(configProps);
 		producerConfig.setCredentialsProvider(AWSUtil.getCredentialsProvider(configProps));
+		producerConfig.setThreadingModel(KinesisProducerConfiguration.ThreadingModel.POOLED);
```

Comment:
Do you think it will make sense to allow the user to configure different threading models?

Reply:
On a second thought for this. Though I believe POOLED model is the best option for most of the use cases I can think of, we should give users the flexibility to make decisions.

Adding PER_REQUEST model